falcon_test.go

```diff
@@ -109,28 +108,232 @@ func TestFalcon(t *testing.T) {
 		// Flip a random bit in the message.
 		badmsg[mathrand.Intn(len(msg))] ^= 1 << mathrand.Intn(8)
 
-		ok = sig.Verify(pub, badmsg)
-		if ok {
-			t.Fatalf("expected verify to fail on modified message")
+		err = pub.Verify(sig, badmsg)
+		if err == nil {
+			t.Fatalf("expected verify to fail on modified message. on pk: %v , sk: %v, msg: %v", pub, priv, msg)
 		}
 
-		badpub := make([]byte, len(pub))
-		copy(badpub, pub)
+		badpub := PublicKey{}
+		copy(badpub[:], pub[:])
 		badpub[mathrand.Intn(len(pub))] ^= 1 << mathrand.Intn(8)
 
-		ok = sig.Verify(badpub, msg)
-		if ok {
-			t.Fatalf("expected verify to fail with modified public key")
+		err = badpub.Verify(sig, msg)
+		if err == nil {
+			t.Fatalf("expected verify to fail with modified public key. on pk: %v , sk: %v, msg: %v", pub, priv, msg)
 		}
 
 		sigCT, err := sig.ConvertToCT()
 		if err != nil {
-			t.Fatal(err)
+			t.Fatalf("failed to conver sign to CT. err: %s on pk: %v , sk: %v, msg: %v", err, pub, priv, msg)
 		}
 
-		ok = sigCT.Verify(pub, msg)
-		if !ok {
-			t.Fatalf("verify_ct failed")
+		err = pub.VerifyCTSignature(sigCT, msg)
+		if err != nil {
+			t.Fatalf("verify_ct failed err msg %s on pk: %v , sk: %v, msg: %v", err, pub, priv, msg)
 		}
 	}
 }
+
+func TestFalconCompressedSignatureSizes(t *testing.T) {
+	seed := make([]byte, 64)
+	rand.Read(seed)
+
+	pub, priv, err := GenerateKey(seed)
+	if err != nil {
+		t.Fatalf("failed to generate keys. err message: %s", err)
+	}
+
+	msg := make([]byte, 500)
+	rand.Read(msg)
+
+	sig, err := priv.SignCompressed(msg)
+	if err != nil {
+		t.Fatalf("failed to sign message. err message: %s", err)
+	}
+
+	var sig2 [SignatureMaxSize + 1]byte
+	copy(sig2[:], sig)
+	err = pub.Verify(sig2[:], msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+
+	var sig3 [1]byte
+	copy(sig3[:], sig)
+	err = pub.Verify(sig3[:], msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
+
+	var sig4 [2]byte
+	copy(sig4[:], sig)
+	err = pub.Verify(sig4[:], msg)
+	if err == nil {
+		t.Fatalf("verification succeeded. should have failed.")
+	}
```

Comment:
the last two could be generalized to testing any SignatureMaxSize-1 bytes out of the signature. Also, need to be extended to slices greater than SignatureMaxSize.